test/pkg/test/framework.go

```diff
@@ -45,21 +49,28 @@ var (
 
 type Framework struct {
 	client.Client
-	LatticeClient services.Lattice
+	LatticeClient                       services.Lattice
+	TestCasesCreatedServiceNetworkNames map[string]bool //key: ServiceNetworkName; value: not in use, meaningless
+	TestCasesCreatedServiceNames        map[string]bool //key: ServiceName; value not in use, meaningless
+	TestCasesCreatedTargetGroupNames    map[string]bool //key: TargetGroupName; value: not in use, meaningless
 }
 
 func NewFramework(ctx context.Context) *Framework {
 	var scheme = scheme.Scheme
 	lo.Must0(v1beta1.Install(scheme))
 	lo.Must0(v1alpha1.Install(scheme))
 	framework := &Framework{
-		Client:        lo.Must(client.New(controllerruntime.GetConfigOrDie(), client.Options{Scheme: scheme})),
-		LatticeClient: services.NewDefaultLattice(session.Must(session.NewSession()), ""), // region is currently hardcoded
+		Client:                              lo.Must(client.New(controllerruntime.GetConfigOrDie(), client.Options{Scheme: scheme})),
+		LatticeClient:                       services.NewDefaultLattice(session.Must(session.NewSession()), config.Region), // region is currently hardcoded
+		TestCasesCreatedServiceNetworkNames: make(map[string]bool),
+		TestCasesCreatedServiceNames:        make(map[string]bool),
+		TestCasesCreatedTargetGroupNames:    make(map[string]bool),
 	}
+
 	SetDefaultEventuallyTimeout(180 * time.Second)
-	SetDefaultEventuallyPollingInterval(1 * time.Second)
-	BeforeEach(func() { framework.ExpectToBeClean(ctx) })
```

Comment:
can we keep this and update the framework.ExpectToBeClean() to only check the lattice resource against current VPC?

Reply:
Should I add `framework.ExpectToBeClean(ctx)` in `BeforeEach()` or `AfterEach()` or `AfterAll()` ?

I prefer add `ExpectToBeClean` in `AfterAll()` in the  httproute_test.go file.

Current ExpectToBeClean() will check whether __current test run__ created K8s resource (current run only, as I changed `DiscoveryLabel`)  AND check whether  __current test run__ created VPCLattice resources are clean, I could change this logic